src/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -32,7 +33,7 @@ internal struct WriteStackFrame
         // The current property.
         public bool PropertyEnumeratorActive;
         public ExtensionDataWriteStatus ExtensionDataStatus;
-        public IEnumerator PropertyEnumerator;
+        public Dictionary<string, JsonPropertyInfo>.Enumerator PropertyEnumerator;
```

Comment:
This changes behavior if `WriteStackFrame is copied: previously `PropertyEnumerator` would all point to the same instance, but now they get a copy.

Not sure how `WriteStackFrame` is used; is this acceptable?

Reply:
I believe all our current usages of `WriteStackFrame` is pass-by-ref, and this struct isn't meant to be copied. We have one place where it's pass by in. @steveharter - can you confirm my assumption that we never intend to copy a `WriteStackFrame`?

Reply:
Lots of things would break if this struct were copied.